Local failing tests

**Describe the issue you encountered:**
Tests that are failing locally (63ff18d)
```
1) Advanced payment panel tests can backup wallet to file:
     WaitUntilTimeoutError: Promise was rejected with the following reason: AssertionError: null == 'file:'
      at getValue("#urlInput") - ledgerPanelAdvancedPanelTest.js:161:21

  2) Advanced payment panel tests can recover wallet from file:
     Error: timeout of 30000ms exceeded. Ensure the done() callback is being called in this test.


  3) Advanced payment panel tests shows an error popover if one recovery key is missing:
     TypeError: path must be a string or Buffer
      at Object.fs.openSync (fs.js:584:18)
      at Object.fs.writeFileSync (fs.js:1316:33)
      at generateAndSaveRecoveryFile (test/about/ledgerPanelAdvancedPanelTest.js:130:6)
      at Context.<anonymous> (test/about/ledgerPanelAdvancedPanelTest.js:199:5)
      at Generator.next (<anonymous>)
      at onFulfilled (node_modules/co/index.js:65:19)
      at node_modules/co/index.js:54:5
      at Context.co (node_modules/co/index.js:50:10)
      at Context.createPromise (node_modules/co/index.js:30:15)
      at Test.Runnable.run (node_modules/co-mocha/lib/co-mocha.js:43:16)
      at process._tickCallback (internal/process/next_tick.js:109:7)

  4) Advanced payment panel tests shows an error popover if a recovery key is not a UUID:
     TypeError: path must be a string or Buffer
      at Object.fs.openSync (fs.js:584:18)
      at Object.fs.writeFileSync (fs.js:1316:33)
      at generateAndSaveRecoveryFile (test/about/ledgerPanelAdvancedPanelTest.js:130:6)
      at Context.<anonymous> (test/about/ledgerPanelAdvancedPanelTest.js:206:5)
      at Generator.next (<anonymous>)
      at onFulfilled (node_modules/co/index.js:65:19)
      at node_modules/co/index.js:54:5
      at Context.co (node_modules/co/index.js:50:10)
      at Context.createPromise (node_modules/co/index.js:30:15)
      at Test.Runnable.run (node_modules/co-mocha/lib/co-mocha.js:43:16)
      at process._tickCallback (internal/process/next_tick.js:109:7)

  5) Advanced payment panel tests shows an error popover if both recovery keys are missing:
     TypeError: path must be a string or Buffer
      at Object.fs.openSync (fs.js:584:18)
      at Object.fs.writeFileSync (fs.js:1316:33)
      at generateAndSaveRecoveryFile (test/about/ledgerPanelAdvancedPanelTest.js:130:6)
      at Context.<anonymous> (test/about/ledgerPanelAdvancedPanelTest.js:213:5)
      at Generator.next (<anonymous>)
      at onFulfilled (node_modules/co/index.js:65:19)
      at node_modules/co/index.js:54:5
      at Context.co (node_modules/co/index.js:50:10)
      at Context.createPromise (node_modules/co/index.js:30:15)
      at Test.Runnable.run (node_modules/co-mocha/lib/co-mocha.js:43:16)
      at process._tickCallback (internal/process/next_tick.js:109:7)

  6) Advanced payment panel tests shows an error popover if the file is empty:
     TypeError: path must be a string or Buffer
      at Object.fs.openSync (fs.js:584:18)
      at Object.fs.writeFileSync (fs.js:1316:33)
      at generateAndSaveRecoveryFile (test/about/ledgerPanelAdvancedPanelTest.js:130:6)
      at Context.<anonymous> (test/about/ledgerPanelAdvancedPanelTest.js:220:5)
      at Generator.next (<anonymous>)
      at onFulfilled (node_modules/co/index.js:65:19)
      at node_modules/co/index.js:54:5
      at Context.co (node_modules/co/index.js:50:10)
      at Context.createPromise (node_modules/co/index.js:30:15)
      at Test.Runnable.run (node_modules/co-mocha/lib/co-mocha.js:43:16)
      at process._tickCallback (internal/process/next_tick.js:109:7)

  7) Regular payment panel tests can setup payments "before each" hook for "shows welcome page":
     WaitUntilTimeoutError: Promise was rejected with the following reason: timeout
      at getUrl() - brave.js:282:40

  8) Ledger table 2 publishers pin excluded publisher:
     WaitUntilTimeoutError: element ([data-tbody-index="1"] [data-row-index="1"] [data-switch-status="false"]) still not visible after 10000ms
      at isVisible("[data-tbody-index="1"] [data-row-index="1"] [data-switch-status="false"]") - waitForVisible.js:37:22

  9) Ledger table 2 publishers check pinned sites amount, when you have 0 eligible unpinned sites:
     Promise was rejected with the following reason: timeout
  Error

  10) MessageBox component tests alert when opening a new tab (while alert is showing) "before each" hook for "lets you use the back button":
     element (#thelink[href="page2.html"]) still not visible after 5000ms
  Error

  11) Clear Browsing Panel with saved state values saves the history switch state:
     Promise was rejected with the following reason: timeout
  Error

  12) Clear Browsing Panel with history clears the browsing history:
     Promise was rejected with the following reason: timeout
  Error

  13) Clear Browsing Panel with closedFrames clears closedFrames:
     Promise was rejected with the following reason: timeout
  Error

  14) noscript can allow scripts when the url host is an ipv6 address:
     WaitUntilTimeoutError: Promise was rejected with the following reason: timeout
      at getText(".blockedOriginsList") - noScriptTest.js:151:23

  15) Sync Panel sync post-setup "before each" hook for "sync can be toggled":
     WaitUntilTimeoutError: Promise was rejected with the following reason: timeout
      at windowHandles() - brave.js:278:32

  16) Syncing and clearing data prevents it from syncing history:
     Promise was rejected with the following reason: timeout
  Error

  17) Syncing and clearing data prevents it from syncing site settings:
     Promise was rejected with the following reason: timeout
  Error

  18) urlBar tests keeps url text separate from suffix text changes only the selection:
     WaitUntilTimeoutError: Promise was rejected with the following reason: timeout
      at getValue("#urlInput") - brave.js:549:23

  19) tab tests webview in fullscreen mode keep favicon and title after exiting fullscreen mode:
     WaitUntilTimeoutError: element ([data-test-active-tab] [data-test-id="tabTitle"]) still not visible after 10000ms
      at isVisible("[data-test-active-tab] [data-test-id="tabTitle"]") - waitForVisible.js:37:22

  20) tab tests tab transfer can detach into new windows:
     Promise was rejected with the following reason: timeout
  Error
 21) ContextMenu navigation check if enter is prevented:
     WaitUntilTimeoutError: element (.contextMenu div.selectedByKeyboard[data-index="0"]) still not existing after 10000ms
      at isExisting(".contextMenu div.selectedByKeyboard[data-index="0"]") - waitForExist.js:37:22
  22) ContextMenu navigation click on item:
     WaitUntilTimeoutError: element (.contextMenu) still not visible after 10000ms
      at isVisible(".contextMenu") - waitForVisible.js:37:22
  23) ContextMenu navigation select item via click and keys:
     WaitUntilTimeoutError: element (.contextMenu) still not visible after 10000ms
      at isVisible(".contextMenu") - waitForVisible.js:37:22
  24) ContextMenu navigation select item via keys only:
     WaitUntilTimeoutError: element (.contextMenu) still not visible after 10000ms
      at isVisible(".contextMenu") - waitForVisible.js:37:22
  25) ContextMenu navigation check left/right on non sub menu item:
     WaitUntilTimeoutError: element (.contextMenu) still not visible after 10000ms
      at isVisible(".contextMenu") - waitForVisible.js:37:22
```

--- a/app/browser/menu.js
+++ b/app/browser/menu.js
@@ -694,7 +694,11 @@ const doAction = (action) => {
       break
     case appConstants.APP_ON_CLEAR_BROWSING_DATA:
       appDispatcher.waitFor([appStore.dispatchToken], () => {
-        if (appStore.getState().getIn(['clearBrowsingDataDefaults', 'browserHistory'])) {
+        const state = appStore.getState()
+        const defaults = state.get('clearBrowsingDataDefaults')
+        const temp = state.get('tempClearBrowsingData', Immutable.Map())
+        const clearData = defaults ? defaults.merge(temp) : temp
+        if (clearData.get('browserHistory')) {
           createMenu()
         }
       })

--- a/app/browser/reducers/sitesReducer.js
+++ b/app/browser/reducers/sitesReducer.js
@@ -42,11 +42,16 @@ const sitesReducer = (state, action, immutableAction) => {
       state = siteCache.loadLocationSiteKeysCache(state)
       break
     case appConstants.APP_ON_CLEAR_BROWSING_DATA:
-      if (state.getIn(['clearBrowsingDataDefaults', 'browserHistory'])) {
-        state = state.set('sites', siteUtil.clearHistory(state.get('sites')))
-        filtering.clearHistory()
+      {
+        const defaults = state.get('clearBrowsingDataDefaults')
+        const temp = state.get('tempClearBrowsingData', Immutable.Map())
+        const clearData = defaults ? defaults.merge(temp) : temp
+        if (clearData.get('browserHistory')) {
+          state = state.set('sites', siteUtil.clearHistory(state.get('sites')))
+          filtering.clearHistory()
+        }
+        break
       }
-      break
     case appConstants.APP_ADD_SITE:
       const isSyncEnabled = syncEnabled()
       if (Immutable.List.isList(action.siteDetail)) {

--- a/app/ledger.js
+++ b/app/ledger.js
@@ -37,6 +37,7 @@ const path = require('path')
 const urlParse = require('./common/urlParse')
 const urlFormat = require('url').format
 const util = require('util')
+const Immutable = require('immutable')
 
 const electron = require('electron')
 const app = electron.app
@@ -179,10 +180,15 @@ const doAction = (state, action) => {
       break
 
     case appConstants.APP_ON_CLEAR_BROWSING_DATA:
-      if (state.getIn(['clearBrowsingDataDefaults', 'browserHistory']) && !getSetting(settings.PAYMENTS_ENABLED)) {
-        reset(true)
+      {
+        const defaults = state.get('clearBrowsingDataDefaults')
+        const temp = state.get('tempClearBrowsingData', Immutable.Map())
+        const clearData = defaults ? defaults.merge(temp) : temp
+        if (clearData.get('browserHistory') && !getSetting(settings.PAYMENTS_ENABLED)) {
+          reset(true)
+        }
+        break
       }
-      break
 
     case appConstants.APP_IDLE_STATE_CHANGED:
       visit('NOOP', underscore.now(), null)

--- a/app/renderer/components/common/contextMenu.js
+++ b/app/renderer/components/common/contextMenu.js
@@ -474,7 +474,7 @@ class ContextMenu extends React.Component {
       ? [0]
       : props.selectedIndex
     props.selectedIndexChild = props.selectedIndex
-      ? this.props.selectedIndex[0]
+      ? props.selectedIndex[0]
       : null
     props.left = contextMenuDetail.get('left')
     props.right = contextMenuDetail.get('right')

--- a/js/stores/appStore.js
+++ b/js/stores/appStore.js
@@ -658,7 +658,7 @@ const handleAppAction = (action) => {
       break
     case appConstants.APP_ON_CLEAR_BROWSING_DATA:
       const defaults = appState.get('clearBrowsingDataDefaults')
-      const temp = appState.get('tempClearBrowsingData')
+      const temp = appState.get('tempClearBrowsingData', Immutable.Map())
       const clearData = defaults ? defaults.merge(temp) : temp
 
       if (clearData.get('browserHistory')) {

--- a/test/about/ledgerTableTest.js
+++ b/test/about/ledgerTableTest.js
@@ -140,8 +140,8 @@ describe('Ledger table', function () {
       yield this.app.client
         .tabByIndex(0)
         .click(`${secondTableFirstRow} .switchBackground`)
-        .waitForVisible(`${secondTableSecondRow} [data-switch-status="false"]`)
-        .click(`${secondTableSecondRow} [data-test-pinned="false"]`)
+        .waitForVisible(`${secondTableFirstRow} [data-switch-status="false"]`)
+        .click(`${secondTableFirstRow} [data-test-pinned="false"]`)
         .waitForVisible(`${firstTableFirstRow} [data-test-pinned="true"]`)
         .windowByUrl(Brave.browserWindowUrl)
         .waitUntilSynopsis(function (synopsis) {

--- a/test/about/newTabTest.js
+++ b/test/about/newTabTest.js
@@ -142,7 +142,7 @@ describe('about:newtab tests', function () {
       })
 
       it('shows sites that have been visited', function * () {
-        yield this.app.client.onClearBrowsingData({browserHistory: true})
+        yield this.app.client.onClearBrowsingData('browserHistory', true)
 
         yield loadPageWithTracker(this.app.client)
 
@@ -153,7 +153,7 @@ describe('about:newtab tests', function () {
       })
 
       it('lets you pin a tile (and shows the pinned icon afterwards)', function * () {
-        yield this.app.client.onClearBrowsingData({browserHistory: true})
+        yield this.app.client.onClearBrowsingData('browserHistory', true)
 
         yield loadPageWithTracker(this.app.client)
 
@@ -169,7 +169,7 @@ describe('about:newtab tests', function () {
       })
 
       it('doesn\'t show about pages on topSites grid', function * () {
-        yield this.app.client.onClearBrowsingData({browserHistory: true})
+        yield this.app.client.onClearBrowsingData('browserHistory', true)
 
         // Adding about pages shouldn't add them to topSites grid
         yield addDemoAboutPages(this.app.client)
@@ -196,7 +196,7 @@ describe('about:newtab tests', function () {
       it('shows favicon image for topSites', function * () {
         const pageWithFavicon = Brave.server.url('favicon.html')
         yield this.app.client
-          .onClearBrowsingData({browserHistory: true})
+          .onClearBrowsingData('browserHistory', true)
           .tabByUrl(Brave.newTabUrl)
           .url(pageWithFavicon)
           .waitForUrl(pageWithFavicon)
@@ -209,7 +209,7 @@ describe('about:newtab tests', function () {
       })
 
       it('replace topSites favicon images with a letter when no icon is found', function * () {
-        yield this.app.client.onClearBrowsingData({browserHistory: true})
+        yield this.app.client.onClearBrowsingData('browserHistory', true)
 
         const pageWithoutFavicon = Brave.server.url('page_favicon_not_found.html')
 

--- a/test/app/sessionStoreTest.js
+++ b/test/app/sessionStoreTest.js
@@ -24,7 +24,7 @@ describe('sessionStore', function () {
       yield setup(Brave.app.client)
       yield Brave.app.client
         .waitForBrowserWindow()
-        .onClearBrowsingData({browserHistory: true})
+        .onClearBrowsingData('browserHistory', true)
         .waitForUrl(Brave.newTabUrl)
         .loadUrl(page1Url)
         .windowParentByUrl(page1Url)

--- a/test/bravery-components/clearBrowsingDataPanelTest.js
+++ b/test/bravery-components/clearBrowsingDataPanelTest.js
@@ -87,7 +87,7 @@ describe('Clear Browsing Panel', function () {
       const page1Url = Brave.server.url('page1.html')
       yield setup(this.app.client)
       yield this.app.client
-        .onClearBrowsingData({browserHistory: true})
+        .onClearBrowsingData('browserHistory', true)
         .tabByIndex(0)
         .loadUrl(page1Url)
         .waitForBrowserWindow()
@@ -117,7 +117,7 @@ describe('Clear Browsing Panel', function () {
         .waitForVisible(clearBrowsingDataButton)
         .click(clearBrowsingDataButton)
         .waitForBrowserWindow()
-        .waitForVisible('[data-test-id="browserHistorySwitch"]')
+        .waitForVisible('[data-test-id="browserHistorySwitch"] .switchedOn')
         .waitForVisible(clearDataButton)
         .click(clearDataButton)
         .waitUntil(function () {
@@ -134,7 +134,7 @@ describe('Clear Browsing Panel', function () {
       const page1Url = Brave.server.url('page1.html')
       yield setup(this.app.client)
       yield this.app.client
-        .onClearBrowsingData({browserHistory: true})
+        .onClearBrowsingData('browserHistory', true)
         .tabByIndex(0)
         .loadUrl(page1Url)
         .waitForBrowserWindow()
@@ -161,7 +161,7 @@ describe('Clear Browsing Panel', function () {
             return val.value.closedFrames.length === 1
           })
         })
-        .onClearBrowsingData({browserHistory: true})
+        .onClearBrowsingData('browserHistory', true)
         .waitUntil(function () {
           return this.getWindowState().then((val) => {
             return val.value.closedFrames.length === 0

--- a/test/lib/brave.js
+++ b/test/lib/brave.js
@@ -781,11 +781,12 @@ var exports = {
      *
      * @param {object} clearDataDetail - the options to use for clearing
      */
-    this.app.client.addCommand('onClearBrowsingData', function (clearDataDetail) {
-      logVerbose('onClearBrowsingData("' + clearDataDetail + '")')
-      return this.execute(function (clearDataDetail) {
-        return devTools('appActions').onClearBrowsingData(clearDataDetail)
-      }, clearDataDetail).then((response) => response.value)
+    this.app.client.addCommand('onClearBrowsingData', function (key, value) {
+      logVerbose('onClearBrowsingData("' + key + ': ' + value + '")')
+      return this.execute(function (key, value) {
+        devTools('appActions').onToggleBrowsingData(key, value)
+        return devTools('appActions').onClearBrowsingData()
+      }, key, value).then((response) => response.value)
     })
 
     this.app.client.addCommand('getDefaultWindowHeight', function () {

--- a/test/misc-components/syncTest.js
+++ b/test/misc-components/syncTest.js
@@ -636,7 +636,7 @@ describe('Syncing history', function () {
     yield setupBrave(Brave.app.client)
     // New tab sites appear in history; so clear them out.
     yield Brave.app.client
-      .onClearBrowsingData({browserHistory: true})
+      .onClearBrowsingData('browserHistory', true)
       .changeSetting(settings.NEWTAB_MODE, newTabMode.EMPTY_NEW_TAB)
       .changeSetting(settings.SYNC_TYPE_HISTORY, true)
     yield setupSync(Brave.app.client, seed)
@@ -780,7 +780,7 @@ describe('Syncing and clearing data prevents it from syncing', function () {
     yield setupBrave(Brave.app.client)
     // New tab sites appear in history; so first clear them out.
     yield Brave.app.client
-      .onClearBrowsingData({browserHistory: true})
+      .onClearBrowsingData('browserHistory', true)
       .changeSetting(settings.NEWTAB_MODE, newTabMode.EMPTY_NEW_TAB)
       .changeSetting(settings.SYNC_TYPE_HISTORY, true)
     yield setupSync(Brave.app.client, seed)
@@ -801,7 +801,8 @@ describe('Syncing and clearing data prevents it from syncing', function () {
       .click(noScriptSwitch)
       .waitUntil(checkSiteSetting(hostPattern, 'noScript', true))
       .pause(1000) // XXX: Wait for sync to upload records to S3
-      .onClearBrowsingData({browserHistory: true, savedSiteSettings: true})
+      .onClearBrowsingData('browserHistory', true)
+      .onClearBrowsingData('savedSiteSettings', true)
       .pause(500) // XXX: Wait for sync to delete records from S3
 
     // Finally start a fresh profile and setup sync
@@ -842,7 +843,7 @@ describe('Syncing then turning it off stops syncing', function () {
     yield setupBrave(Brave.app.client)
     // New tab sites appear in history; so first clear them out.
     yield Brave.app.client
-      .onClearBrowsingData({browserHistory: true})
+      .onClearBrowsingData('browserHistory', true)
       .changeSetting(settings.NEWTAB_MODE, newTabMode.EMPTY_NEW_TAB)
       .changeSetting(settings.SHOW_BOOKMARKS_TOOLBAR, true)
       .changeSetting(settings.SYNC_TYPE_HISTORY, true)

--- a/test/navbar-components/urlBarTest.js
+++ b/test/navbar-components/urlBarTest.js
@@ -27,7 +27,7 @@ describe('urlBar tests', function () {
       yield this.app.client.waitForExist(urlInput)
       yield this.app.client.waitForElementFocus(urlInput)
       yield this.app.client
-        .onClearBrowsingData({browserHistory: true})
+        .onClearBrowsingData('browserHistory', true)
         .addSite({ location: 'https://brave.com', title: 'Brave' })
     })
 
@@ -63,7 +63,7 @@ describe('urlBar tests', function () {
         .waitForInputText(urlInput, '')
 
       yield this.app.client
-        .onClearBrowsingData({browserHistory: true})
+        .onClearBrowsingData('browserHistory', true)
         .addSite({ location: 'https://brave.com', title: 'Brave' })
         .addSite({ location: 'https://brave.com/test' })
         .addSite({ location: 'https://www.youtube.com' })
@@ -760,7 +760,7 @@ describe('urlBar tests', function () {
       yield this.app.client.waitForExist(urlInput)
       yield this.app.client.waitForElementFocus(urlInput)
       yield this.app.client
-        .onClearBrowsingData({browserHistory: true})
+        .onClearBrowsingData('browserHistory', true)
         .addSite({ location: 'https://github.com/brave/browser-laptop', title: 'browser-laptop' })
         .addSite({ location: 'https://github.com/brave/ad-block', title: 'Muon' })
     })

